packages/sign_in_with_apple/sign_in_with_apple/CHANGELOG.md

```diff
@@ -1,3 +1,8 @@
+## 4.4.0
+
+- Support Gradle 8 ([@davidmartos96](https://github.com/davidmartos96) in [#375](https://github.com/aboutyou/dart_packages/pull/375))
+  - Requires at least Flutter 3.7.12 which brings full Gradle 8 support
```

Comment:
TBD if we should then bump to version 5.0.0. I think it's fine an consumers will just pick the highest package they can support.

Or maybe we find a way to keep this fully compatible with Flutter 2/3.x, but I was unable to get that working locally.

Reply:
Are you trying in the example app? Or a new project?
Maybe this is relevant? https://docs.flutter.dev/release/breaking-changes/android-java-gradle-migration-guide

Reply:
Which particular 2.x and 3.x versions are not working? I assume 2.8.0 since it's the minimum stated in the pubspec

Reply:
> Are you trying in the example app? Or a new project?

I was looking at the example app.

I tried 2.8, 3.0, and 3.3 (all in their latest patch releases).

But I suppose it might be related to my locally upgraded Android Studio + Java version.

Maybe there is a way to keep better backwards compatibility, I'll check this against the migrations we did in the production app itself.

Reply:
I'll try 2.8 and 3.3 with previous Android Studio (Eel). That could be the problem indeed.
I'm not sure there is a backwards compatibility fix when using AS Flamingo, given that Flutter has this in their home page (https://docs.flutter.dev/release/breaking-changes/android-java-gradle-migration-guide)

Reply:
@tp  I managed to run it correctly with Flutter 2.8 and 3.3.
What I did:

* Install Android Studio Electric Eel (https://developer.android.com/studio/archive)
* Remove the recently added >=3.7 constraint
*  Keep the `androidx.browser:browser` dependency in 1.4.0. Before this PR it was 1.3.0 and latest is 1.5.0 but 1.5.0 requires android SDK 33. Flutter 2.8 and 3.3 by default, when installing, use SDK 31. I'm Flutter < 3.7 supports SDK 33. 
Changelog for the browser library: https://developer.android.com/jetpack/androidx/releases/browser

Reply:
I run the example app by opening the `example/android` folder with this Android Studio. I believe a special flag for `flutter run` is needed to provide a different android studio installation, so I found easier to do `flutter pub get` and run from Android Studio directly

Reply:
@davidmartos96 Oh nice, good to know that it'd still work with a legacy Android Studio.

Question is to we really want to put in the extra effort to keep this supported, or maybe take this as a good time to require at least e.g. Flutter 3.7.6 which most people should be using by now anyway. Unless someone chimes in here with a specific backwards-compatibility requirement, I think it would be beneficial to ease our support burden here.

Reply:
SDK 33 will be a requirement in August. I see 2 options.
1. Update the constraint to 3.7.6 now and update the browser dependency to 1.5.0
2. Keep the 2.8.0 constraint with browser 1.4.0 and in a few months update it, since SDK 33 (Flutter 3.7) will be a requirement.

Reply:
Source https://support.google.com/googleplay/android-developer/answer/11926878

Reply:
Yes, same thoughts.
Should the browser dependency with updated to 1.5.0 on Android? 1.3.0 is from December 2, 2020

Reply:
It is already, right?

https://github.com/aboutyou/dart_packages/blob/71741b933bc3d0846af1178f7ee41e0f0e959b55/packages/sign_in_with_apple/sign_in_with_apple/android/build.gradle#L58

Reply:
@tp 😅 Totally right, testing it with the old AS confused me haha

PR changes LGTM!